internal/app/upstream/multiplexer.go

```diff
@@ -0,0 +1,34 @@
+package multiplexer
+
+import (
+	"context"
+
+	v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	"google.golang.org/grpc"
+)
+
+// Multiplexer handles the connections to the upstream control plane
+type Multiplexer interface {
+	// QueueRequest starts a stream with the upstream control plane
+	QueueRequest(context.Context, chan *v2.DiscoveryRequest, chan *v2.DiscoveryResponse)
```

Comment:
Do calls to this function always succeed?

Reply:
Yes..The function will do async calls and retries internally. I'm still reading up on how to handle errors and bubble them up when using channels.

Reply:
found that using an error pipeline is a wellknown pattern , [reading](https://blog.golang.org/pipelines),
[followup reading](https://medium.com/statuscode/pipeline-patterns-in-go-a37bb3a7e61d)

Reply:
QueueRequest now sends an error.